CRM/Volunteer/Form/Defaults.php

```diff
@@ -114,18 +106,29 @@ function validate() {
   function postProcess() {
     $values = $this->exportValues();
 
-    CRM_Core_BAO_Setting::setItem($values['volunteer_default_profiles_individual'],"volunteer_defaults", "volunteer_default_profiles_individual");
-    CRM_Core_BAO_Setting::setItem($values['volunteer_default_profiles_group'],"volunteer_defaults", "volunteer_default_profiles_group");
-    CRM_Core_BAO_Setting::setItem($values['volunteer_default_profiles_both'],"volunteer_defaults", "volunteer_default_profiles_both");
-    CRM_Core_BAO_Setting::setItem($values['volunteer_default_campaign'],"volunteer_defaults", "volunteer_default_campaign");
-    CRM_Core_BAO_Setting::setItem($values['volunteer_default_locblock'],"volunteer_defaults", "volunteer_default_locblock");
 
-    $isActive = array_key_exists("volunteer_default_is_active", $values) ? $values['volunteer_default_is_active'] : 0;
-    CRM_Core_BAO_Setting::setItem($isActive, "volunteer_defaults", "volunteer_default_is_active");
+    //Compose the profiles before we save tem.
+    $profiles = array();
+
+    foreach(CRM_Volunteer_BAO_Project::getProjectProfileAudienceTypes() as $audience) {
+      $profiles[$audience['type']] = CRM_Utils_Array::value('volunteer_default_profiles_' . $audience['type'], $values);
```

Comment:
Do we need to do any type checking here? Seems like there's a chance we'll get an empty string or some other falsey value we want to represent as an empty array...